examples/seismic/model.py

```diff
@@ -52,53 +53,96 @@ def initialize_damp(damp, nbl, spacing, abc_type="damp"):
 
 class PhysicalDomain(SubDomain):
 
-    name = 'phydomain'
+    name = 'physdomain'
 
-    def __init__(self, nbl):
+    def __init__(self, so, fs=False):
         super(PhysicalDomain, self).__init__()
-        self.nbl = nbl
+        self.so = so
+        self.fs = fs
+
+    def define(self, dimensions):
+        map_d = {d: d for d in dimensions}
+        if self.fs:
+            map_d[dimensions[-1]] = ('middle', self.so, 0)
+        return map_d
+
+
+class FSDomain(SubDomain):
+
+    name = 'fsdomain'
+
+    def __init__(self, so):
+        super(FSDomain, self).__init__()
+        self.size = so
 
     def define(self, dimensions):
-        return {d: ('middle', self.nbl, self.nbl) for d in dimensions}
+        """
+        Definition of the top part of the domain for wrapped indices FS
+        """
+        z = dimensions[-1]
+        map_d = {d: d for d in dimensions}
+        map_d.update({z: ('left', self.size)})
+        return map_d
```

Comment:
`return {d: (d if not d==dimensions[-1] else ('left', self.size)) for d in dimensions}`?

Reply:
fixed